src/azure-openapi-validator/rules/DescriptionMustNotBeNodeName.ts

```diff
@@ -0,0 +1,29 @@
+import { rules } from "../rule";
+
+rules.push({
+  id: "M2065",
+  name: "DescriptionMustNotBeNodeName",
+  severity: "error",
+  category: ["RPCViolation"],
+
+  appliesTo_JsonQuery: "$..*[?(@.description)]",
+  run: function* (doc, node, path) {
+    let msg: string = "Description must not match the name of the node it is supposed to describe";
+    let nodeName = <any>path[path.length - 1];
+
+    if (!isNaN(nodeName)) {
+      if ((<string>node.name).toLowerCase() === TrimDescription(node.description)) {
+        yield { message: msg + "Node name:'" + node.name + "' " + "Description: '" + node.description + "'", location: path.concat(['description']) };
```

Comment:
The js string doesn't have anything in-built for this unfortunately, if it really is that important we could write an extension method of sorts.